gamersnet_backend/persistence/avatar.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+let ObjectID = require('mongodb').ObjectId;
+let MongoDB = require('./mongodb');
+let db, avatar;
+
+async function connect() {
+    db = await MongoDB.open();
+
+    avatar = db.collection('avatar');
+}
+
+async function upsertAvatar(id, imageBuffer, mimeType) {
+    await connect();
+
+    let result = avatar.updateOne(
+        {userID: id},
+        {
+            $set: {
+                userID: id,
+                contentType: mimeType,
+                avatar: imageBuffer
+            }
+        },
+        {upsert: true}
```

Comment:
Do we want to create a new avatar if an avatar for that user doesn't already exist while updating avatar? Are we using this to create the avatar as well?

Reply:
Currently, the behaviour when the user does not have a profile avatar is to serve a default profile avatar from a file named `defaultAvatar.png` in the `routes/users` directory. It does not insert a new document if one does not exist when fetching the avatar, only when updating.